### ISSUE TITLE
Usability improvement: Select features in freehand mode, polygon mode or radius mode

### ISSUE THREAD

**Issue by tschmetzer** [qgis/QGIS#41622]
Currently the user needs to click twice when graphically selecting features. This doesn't seem much of a diffence yet for dozens, hundrets or even thousend objects it is very frustrating. Suggestion: Start drawing the selection area when mouse button is hold/pressed and select when mouse button is released 

-> Advantage: The user can select single/individual objects by clicking only once instead of twice or three times. If the current selection method should be kept the new suggestion could be implemented as an additional option when pressing a key before selecting or something similar

**Comment by uclaros** [comment]
While this would be handy in some situations, I can see the following drawbacks:
- It will be difficult to type in the Selection radius widget.
- It will not be possible to implement in the select by polygon mode.

**Comment by tschmetzer** [comment]
> While this would be handy in some situations, I can see the following drawbacks:
>     * It will be difficult to type in the Selection radius widget.
Good point! 

>     * It will not be possible to implement in the select by polygon mode.

What's the reason behind that? Especially for freehand and polygon mode I would consider it very valuable.

**Comment by uclaros** [comment]
Freehand is doable, but polygon requires distinct mouse clicks for vertices!

**Comment by tschmetzer** [comment]
You're totally right, of course several distinct mouse clicks are required. I was thinking of a more "natural" way to draw these lines as I am somehow used to press the mouse button down and drag the line. But then the question is how to continue setting the next vertices and finish the line which is where it gets difficult. I have two ideas:
- Putting a vertex whenever a mouse button is released
- Putting a vertex whenever a mouse button is released or pressed down (which might be awkward at the beginning but mbaye also very comfortable; without trying it it's hard to imagine what it might feel like)

Independant of the mode regarding finishing a selection line: Additionally to a right click as implemented right now a double click could also be used.

**Comment by tschmetzer** [comment]
How funny, I just found an example implementation of the method I was referring to! It can be found on every Windows in MS Paint:

![2021-02-22 11_38_58-add_vertex_filled2 png - Paint](https://user-images.githubusercontent.com/67278094/108697112-b57a2f00-7502-11eb-9948-21e413b8f8c2.png)

**Comment by tschmetzer** [comment]
Point is that there is a inconsistency the user stumbles upon using various selection tools: When using the rectangle selection tool the user needs to click and drag in order to define the selection area whereas when using a different selection tool he needs to click and let go the mouse button before he is able to drag and define the selection area. This inconsistent behaviour is very confusing to the user. We should decide on *one* behaviour throughout the application. We also need to keep in mind that the measure tool follows the same behaviour and needs to be kept consistent as well.
